actor/src/main/scala/org/apache/pekko/io/dns/internal/TcpDnsClient.scala

```diff
@@ -49,6 +48,7 @@ import pekko.util.ByteString
       log.debug("Connected to TCP address [{}]", ns)
       val connection = sender()
       context.become(ready(connection))
+      context.watch(connection)
```

Comment:
Other usages of that connection also watch it to fail on termination:
https://github.com/apache/incubator-pekko/blob/46e60a61fbabce5e3f36a408bfa3d1fb249eef44/stream/src/main/scala/org/apache/pekko/stream/impl/io/TcpStages.scala#L295-L296

That's why I did the same here for a low risk fix of the issue.
In general it would make sense to rethink the construct as a whole as the code is more or less unchanged since at least 2013.